File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def intToBin(arr):
    binarr = [[] for _ in range(len(arr))]
    for idx in range(len(arr)):
        binarr[idx] = bin(arr[idx])
    for idx in range(len(binarr)):
        binarr[idx] = list(binarr[idx][2:])
        if len(binarr[idx]) != len(arr):
            while len(binarr[idx]) != len(arr):
                binarr[idx].insert(0,'0')
    return binarr

def solution(n, arr1, arr2):
    arr1 = intToBin(arr1)
    arr2 = intToBin(arr2)
    ans = []
    answer = [[] for _ in range(len(arr1))]
    for i in range(len(arr1)):
        tmp = ''
        for j in range(len(arr1)):
            if arr1[i][j] == '1' or arr2[i][j] == '1': #벽
                tmp += "#"
            else:
                tmp += " "
        ans.append(tmp)
    return ans

solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28])

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function convertToBin(n, char) {
    var bin = char.toString(2)
    while (bin.length != n) {
        bin = '0' + bin;
    }
    return bin
}

function assembleMap(n, numMap1, numMap2) {
    var secretMap = [];
    
    for (let i = 0; i < n; i++) {
        var inputText = "";
        for (let j = 0; j < n; j++) {
            if (numMap1[i][j] == 1 || numMap2[i][j] == 1) {
                inputText += "#";
            } else {
                inputText += " ";
            }
        }
        secretMap.push(inputText)
    }
    return secretMap;
}

function solution(n, arr1, arr2) {
    var answer = [];
    var temp1 = [];
    var temp2 = [];
    for (let i = 0; i < n; i++) {
        temp1.push(convertToBin(n, arr1[i]))
        temp2.push(convertToBin(n, arr2[i]))
    }
    return assembleMap(n, temp1, temp2);
}